security/base.py

```diff
@@ -14,17 +14,63 @@
 #    under the License.
 
 import abc
-import collections
+import datetime
 
 import six
 
 
-Issue = collections.namedtuple("Issue", ["type", "description", "subject"])
+class Issue(object):
+    __slots__ = ("type", "id", "description", "subject", "discovered_at",
+                 "confirmed_at", "resolved_at")
+
+    def __init__(self, _type, description, subject, discovered_at=None,
+                 confirmed_at=None, resolved_at=None, **kwargs):
+        self.type = _type
+        self.description = description
+        self.subject = subject
+
+        self.discovered_at = discovered_at
+        self.confirmed_at = confirmed_at
+        self.resolved_at = resolved_at
+
+        for field in ("discovered_at", "confirmed_at", "resolved_at"):
+            value = getattr(self, field)
+            if isinstance(value, datetime.datetime):
+                setattr(self, field, value.isoformat("T"))
+
+        self.id = subject["id"]
+        if (self.confirmed_at is not None
+                and self.confirmed_at < self.discovered_at):
```

Comment:
so we are accepting in this method discovered_at = None, so it may happen that discovered_at is None, either check discovered_at for non or make this argument required.